common/base/StrError_R.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * StrError_R.cpp
+ * Helper function for StrError_R_XSI().
+ * Copyright (C) 2018 Shenghao Yang
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif  // HAVE_CONFIG_H
+
+#include <stdio.h>
+#include <string.h>
+
+#include <limits>
+#include <string>
+#include <sstream>
+
+#include "ola/base/StrError_R.h"
+
+namespace ola {
+
+const int StrError_R_BufSize = 1024;
+
+std::string StrError_R(int errnum) {
+  // Pre-allocate to prevent re-allocation.
+  std::string out(StrError_R_BufSize, '\0');
+  if (StrError_R_XSI(errnum, &(out[0]), out.size())) {
+    std::ostringstream errs("errno = ");
```

Comment:
Sorry, I realised after I said this, I should just get you to use our existing function to do this ola::strings::IntToString :
http://docs.openlighting.org/ola/doc/latest/namespaceola_1_1strings.html#a1c6637c4498627e405cf8c95cef677c7